src/main/kotlin/com/openlattice/entitysets/controllers/EntitySetsController.kt

```diff
@@ -189,9 +191,73 @@ constructor(
                 .map { AuthorizationUtils.getLastAclKeySafely(it) }
                 .toSet()
 
-        return  entitySetManager.getEntitySetsAsMap(entitySetIds).values.toSet()
+        return entitySetManager.getEntitySetsAsMap(entitySetIds).values.toSet()
     }
 
+    @Timed
+    @RequestMapping(
+            path = [BY_ID_PATH],
+            method = [RequestMethod.POST],
+            consumes = [MediaType.APPLICATION_JSON_VALUE],
+            produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getEntitySetsById(@RequestBody entitySetIds: Set<UUID>): Map<UUID, EntitySet> {
+
+        entitySetIds.forEach { entitySetId -> ensureReadAccess(AclKey(entitySetId)) }
+
+        val entitySets = entitySetManager.getEntitySetsAsMap(entitySetIds)
+
+        val now = OffsetDateTime.now()
+        val events = Lists.newArrayList<AuditableEvent>()
```

Comment:
You could also use `withExpectedSize` here since you know how many events there will be. Or you could just do 
```
val events = entitySets.map{ ... }.toList()`
```